devito/finite_differences/differentiable.py

```diff
@@ -0,0 +1,236 @@
+import sympy
+import numpy as np
+
+from devito.symbolics.search import retrieve_functions
+from devito.symbolics.extended_sympy import FrozenExpr
+
+__all__ = ['Differentiable']
+
+
+class Differentiable(FrozenExpr):
+    """
+    This class represents Devito differentiable objects such as
+    sum of functions, product of function or FD approximation and
+    provides FD shortcuts for such expressions
+    """
+    _op_priority = 100.0
+    is_Function = True
```

Comment:
is it a Function?

Reply:
It's technically not, there for dse debug mostly not entirely sure it's needed